### ISSUE TITLE
Accessibility: Functional icon missing role and accessible name

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51827]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The icon acts as a button but does not have the required role or an accessible name. Screen readers are skipping the icon. The icon is also not keyboard-focusable, meaning keyboard-only or screen reader users cannot access this functionality.

The button also does not have any way of communicating to screen reader users in it is currently pressed (or not).

![T002_ShowHideIconDoNotRoleName](https://github.com/user-attachments/assets/f99115d1-d45a-495c-a647-4c1e4913449d)

### Component or Page: 

Front-end Login Form on My Account Page

### Name affected element:

Show/Password icon

### Relevant code snippets:

`<span class="show-password-input"></span>`

### Recommended fix:

Interactive elements need to be coded with HTML <button> tags so that they can be focused and triggered with a keyboard alone. This element needs to be re-coded to be contained in a button tag. Ensure the button has an aria-label or screen reader only text that accurately describes its purpose.
 
 If you cannot change this to be contained in a <button> tag, it is possible to remediate a div or other container around the icon to function as a button with ARIA. The div would need to be given a button role and tabindex=""0"" to ensure that it can be focused with the keyboard. You would also need to write JavaScript to ensure that the div will function properly as a button for keyboard users and can be triggered with the space bar in addition to the return/enter key. For more information on turning a div into a button, see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
 
 This button also needs an aria-pressed attribute to communicate to screen reader users if their password is currently hidden or visible.

**Comment by darrenjacoby** [comment]
@vikrampm1 this looks like a duplicate of https://github.com/woocommerce/woocommerce/issues/51826, once the `<span>` tag is changed to a `<button>` it should fulfil these requirements.

cc @dkotter

**Comment by vikrampm1** [comment]
@darrenjacoby when you have a PR ready, could you link it to both issues? These are also being tracked in a doc, and I think it would be easier that way.
